src/components/CartCard.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  useCallback,
+  useEffect,
+} from 'react';
+
+import {
+  deleteUserCartProduct,
+  postUserWishlist,
+  updateUserCartProduct,
+} from 'apis';
+import { useLogin } from 'hooks';
+import { CartProductInvalidatorProp } from 'interfaces';
+import throttle from 'lodash.throttle';
+import { useNavigate } from 'react-router-dom';
+import { CartAction } from 'types';
+import { toastError } from 'utils';
+
+export const CartCard = ({
+  product: {
+    id,
+    tags,
+    publisher,
+    name,
+    author,
+    image,
+    imgAlt,
+    description,
+    discountPrice,
+    price,
+    discount,
+    inStock,
+    rating,
+    sellerId,
+    qty,
+  },
+  isInWishlist,
+  invalidate,
+}: CartProductInvalidatorProp) => {
+  const {
+    isAuth,
+    loginUser: { encodedToken },
+  } = useLogin();
+  const navigate = useNavigate();
+
+  const updateQty = async (
+    id: string,
+    action: CartAction,
+    payload: number,
+    encodedToken: string,
+    invalidate: () => void
+  ) => {
+    try {
+      await updateUserCartProduct(id, action, payload, {
+        authorinzation: encodedToken,
+      });
+      invalidate();
+    } catch (error) {
+      toastError();
+    }
+  };
+
+  const removeFromCart = async (
+    id: string,
+    encodedToken: string,
+    invalidate: () => void
+  ) => {
+    try {
+      await deleteUserCartProduct(id, { authorinzation: encodedToken });
+
+      invalidate();
+    } catch (error) {
+      toastError();
+    }
+  };
+  const wishlistToggle = async (
+    id: string,
+    encodedToken: string,
+    invalidate: () => void
+  ) => {
+    try {
+      postUserWishlist(id, { authorinzation: encodedToken });
```

Comment:
Typo here. *authorization.

Reply:
Hahaha this is really funny
this is spilled all over the code and this is why it works fine everywhere. will fix it in the next pr now.

Reply:
I fixed it there